packages/core/src/consumed-thing.ts

```diff
@@ -471,7 +471,7 @@ export default class ConsumedThing extends TD.Thing implements IConsumedThing {
                     this.getSecuritySchemes(form.security),
                     this.getServient().retrieveCredentials(this.id)
                 );
-            } else if (this.security && Array.isArray(this.security) && this.security.length > 0) {
+            } else if (this.security != null && Array.isArray(this.security) && this.security.length > 0) {
```

Comment:
If I am not mistaken, the `null` check for `this.security` is actually not necessary:

```suggestion
            } else if (Array.isArray(this.security) && this.security.length > 0) {
```